engine/flink/process/src/test/scala/pl/touk/nussknacker/engine/process/typeinformation/TypingResultAwareTypeInformationDetectionSpec.scala

```diff
@@ -0,0 +1,93 @@
+package pl.touk.nussknacker.engine.process.typeinformation
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream}
+
+import org.apache.flink.api.common.ExecutionConfig
+import org.apache.flink.api.common.typeinfo.TypeInformation
+import org.apache.flink.api.common.typeutils.TypeSerializerSchemaCompatibility
+import org.apache.flink.core.memory.{DataInputViewStreamWrapper, DataOutputViewStreamWrapper}
+import org.scalatest.{FunSuite, Matchers}
+import pl.touk.nussknacker.engine.api.{Context, ValueWithContext}
+import pl.touk.nussknacker.engine.api.context.ValidationContext
+import pl.touk.nussknacker.engine.api.typed.typing.{Typed, TypedObjectTypingResult, TypingResult}
+
+class TypingResultAwareTypeInformationDetectionSpec extends FunSuite with Matchers {
```

Comment:
I wonder if you can add some tests showing problems that can occur during using this dection. E.g. you mention about Int vs Long. What about tests for `nonEqualKeysCompatible` feature?

Reply:
I added additional tests